### PR TITLE
update auto-random.md

### PR DESCRIPTION
**Pull request by qiancai**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)

Refine the description for reserved bits. Close #17826.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v8.2 (TiDB 8.2 versions)
- [x] v8.1 (TiDB 8.1 versions)
- [x] v8.0 (TiDB 8.0 versions)
- [x] v7.6 (TiDB 7.6 versions)
- [x] v7.5 (TiDB 7.5 versions)
- [x] v7.1 (TiDB 7.1 versions)
- [x] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/17908
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by tangenta — APPROVED** [review]

File: auto-random.md [hunk 1]
```diff
@@ -83,16 +83,24 @@ The `AUTO_RANDOM(S, R)` column value automatically assigned by TiDB has a total
...
-| Total number of bits | Sign bit | Reserved bits | Shard bits | Auto-increment bits |
-|---------|---------|-------------|--------|--------------|
-| 64 bits | 0/1 bit | (64-R) bits | S bits | (R-1-S) bits |
+| Signed bit | Reserved bits | Shard bits | Auto-increment bits |
+|---------|-------------|--------|--------------|
+| 1 bit | `64-R` bits | `S` bits | `R-1-S` bits |
 
+The structure of an `AUTO_RANDOM` value without a signed bit is as follows:
+
+Reserved bits | Shard bits | Auto-increment bits |
```
Comment (hfxsd) [inline auto-random.md:94]:
```suggestion
| Reserved bits | Shard bits | Auto-increment bits |
```

File: auto-random.md [hunk 2]
```diff
@@ -104,7 +112,7 @@ The structure of an `AUTO_RANDOM` value is as follows:
 > Selection of range (`R`):
 >
 > - Typically, the `R` parameter needs to be set when the numeric type of the application cannot represent a full 64-bit integer.
-> - For example, the range of JSON number is `[-2^53+1, 2^53-1]`. TiDB can easily assign an integer outside this range to a column of `AUTO_RANDOM(5)`, causing unexpected behaviors when the application reads the column. In this case, you can replace `AUTO_RANDOM(5)` with `AUTO_RANDOM(5, 54)` and TiDB does not assign an integer greater than `9007199254740991` (2^53-1) to the column.
+> - For example, the range of JSON number is `[-(2^53)+1, (2^53)-1]`. TiDB can easily assign an integer beyond this range to a column defined as `AUTO_RANDOM(5)`, causing unexpected behaviors when the application reads the column. In such cases, you can replace `AUTO_RANDOM(5)` with `AUTO_RANDOM(5, 54)` for signed columns and replace `AUTO_RANDOM(5)` with `AUTO_RANDOM(5, 53)` for unsigned columns, ensuring that TiDB does not assign integers greater than `9007199254740991` (2^53-1) to the column.
```
Comment (hfxsd) [inline auto-random.md:115]:
```suggestion
> - For example, the range of JSON number is `[-(2^53)+1, (2^53)-1]`. TiDB can easily assign an integer beyond this range to a column defined as `AUTO_RANDOM(5)`, causing unexpected behaviors when the application reads the column. In such cases, you can replace `AUTO_RANDOM(5)` with `AUTO_RANDOM(5, 54)` for signed columns, and replace `AUTO_RANDOM(5)` with `AUTO_RANDOM(5, 53)` for unsigned columns, ensuring that TiDB does not assign integers greater than `9007199254740991` (2^53-1) to the column.
```

**Review by hfxsd — APPROVED** [review]
Rest LGTM

**Pull request merged**